firedrake/petsc.py

```diff
@@ -3,7 +3,8 @@
 import subprocess
 import petsc4py
 import sys
-petsc4py.init(sys.argv)
+from pyop2.mpi import COMM_WORLD
```

Comment:
```suggestion
from mpi4py.MPI import COMM_WORLD
```
I think we actually want to use the "public" `COMM_WORLD` here so that petsc4py is initialised on the same communicator as a user might use in a script. We do not need to worry about communication clash on the communicator as we know that PETSc is duplicating the communicator behind the scenes anyway.